easybuild/easyblocks/o/openblas.py

```diff
@@ -44,3 +61,42 @@ def sanity_check_step(self):
             'dirs': [],
         }
         super(EB_OpenBLAS, self).sanity_check_step(custom_paths=custom_paths)
+    
+    def _parse_optarch( self, cfg_optarch):
+        """ Plucks the march out of the optarch settings. See
+        https://easybuild.readthedocs.io/en/latest/Controlling_compiler_optimization_flags.html """
+
+        pieces = cfg_optarch.split('=')
+        if len(pieces) > 1:
+            if pieces[0] == 'march':
+                return pieces[1]
+            elif pieces[2] == '-march':
+                return pieces[3]
+        else:
+            return cfg_optarch
+
+    def _get_openblas_targets(self, targetfile):
+        """ Parse the openblas target file and generate a list of targets """
+        targets = []
+
+        with open( targetfile ) as f:
+            lines = f.readlines()
+            for line in lines:
+                # Assumption, the OpenBLAS TargetList.txt has one target per line and that
+                # single words on a line is a target, unless they start with a number and a dot...
+                fields = line.split()
+                if len(fields) == 1:
+                    self.log.debug("Adding OpenBLAS target: "  + fields[0])
+                    targets.append( fields[0] )
+        
+        return targets
+
+    def _get_mapped_target( self, cfg_optarch, openblas_targets ):
+        """ Attempts to find the desired optarch in the list of openblas targets.
+        Returns None if the target was not found. """
+
+        for target in openblas_targets:
+            if cfg_optarch.lower() == target.lower():
+                return target
```

Comment:
We usually try to avoid inline `return`s (they make it harder to follow logic).

```python
result = None
for target in openblas_targets:
    if cfg_optarch.lower() == target.lower():
        result = target

return result

Reply:
Fixed in https://github.com/easybuilders/easybuild-easyblocks/pull/1946/commits/d4da6d55bc8cdfe8534cfd4455adf07adcc5d972